asserts/database.go

```diff
@@ -108,3 +120,44 @@ func (db *Database) ImportKey(authorityID string, privKey *packet.PrivateKey) (f
 	}
 	return fingerp, nil
 }
+
+// use a generalized matching style along what PGP does where keys can be
+// retrieved by giving suffixes of their fingerprint
+// TODO: may need more details about the kind of key we are looking for
+func (db *Database) findPublicKeys(authorityID, fingerprintSuffix string) []PublicKey {
+	res := make([]PublicKey, 0, 1)
+	cands := db.cfg.TrustedKeys[authorityID]
+	for _, cand := range cands {
+		if strings.HasSuffix(cand.Fingerprint(), fingerprintSuffix) {
+			res = append(res, cand)
+		}
+	}
+	// TODO: consider other stored public key assertions
+	return res
+}
+
+// Check tests whether the assertion is properly signed and consistent with all the stored knowledge.
+func (db *Database) Check(assert Assertion) error {
+	content, signature := assert.Signature()
+	sig, err := parseSignature(signature)
+	if err != nil {
+		return err
+	}
+	// TODO: later may need to consider type of assert to find candidate keys
+	pubKeys := db.findPublicKeys(assert.AuthorityID(), sig.KeyID())
+	now := time.Now()
+	err = nil
```

Comment:
Can we please have logic around a `var firstErr error` here?  It will probably look more clear, and avoid a few eye-opening details such as the need to reset err here, and having an error when there are no errors below.